openedx_learning/apps/authoring/units/api.py

```diff
@@ -0,0 +1,218 @@
+"""Units API.
+
+This module provides functions to manage units.
+"""
+
+from django.db.transaction import atomic
+
+from openedx_learning.apps.authoring.containers.models import EntityListRow
+from ..publishing import api as publishing_api
+from ..containers import api as container_api
+from .models import Unit, UnitVersion
+from django.db.models import QuerySet
+
+
+from datetime import datetime
+
+__all__ = [
+    "create_unit",
+    "create_unit_version",
+    "create_next_unit_version",
+    "create_unit_and_version",
+    "get_unit",
+    "get_unit_version",
+    "get_latest_unit_version",
+    "get_user_defined_list_in_unit_version",
+    "get_initial_list_in_unit_version",
+    "get_frozen_list_in_unit_version",
+]
+
+
+def create_unit(
+    learning_package_id: int, key: str, created: datetime, created_by: int | None
+) -> Unit:
+    """Create a new unit.
+
+    Args:
+        learning_package_id: The learning package ID.
+        key: The key.
+        created: The creation date.
+        created_by: The user who created the unit.
+    """
+    with atomic():
+        container = container_api.create_container(
+            learning_package_id, key, created, created_by
+        )
+        unit = Unit.objects.create(
+            container_entity=container,
+            publishable_entity=container.publishable_entity,
+        )
+    return unit
+
+
+def create_unit_version(
+    unit: Unit,
+    version_num: int,
+    title: str,
+    publishable_entities_pks: list[int],
+    draft_version_pks: list[int | None],
+    published_version_pks: list[int | None],
```

Comment:
Why do we specify three different lists of the components in this new unit version? I would naively expect one list of components [and their versions?].

Reply:
I have changed this in my PR.